### ISSUE TITLE
Question about send_file

### ISSUE THREAD

**Issue by kllr** [pgjones/quart#55]
Hi,

I'm trying to generate an excel file in memory and then make it available to download in the browser.
I looked up how to do that in flask and came to the following code:

```
 output = io.BytesIO()
    writer = pd.ExcelWriter(output, engine='xlsxwriter')

    s = pd.DataFrame(statistics.get_all_values_for_download(survey_id))
    s.to_excel(writer, sheet_name='Sheet1')
    writer.save()
    output.seek(0)
    survey = Survey.get(survey_id)
  
    filename = survey.description + '-' + survey.title + datetime.datetime.now().strftime("%d-%b %H:%M")
    return await send_file(output, attachment_filename='{}.xlsx'.format(filename), as_attachment=True)
```

This is giving the error: 
> TypeError expected str, bytes or os.PathLike object, not _io.BytesIO

Reducing it to a simpler form: save the file to disk and serving it (and maybe removing it with a cronjob) seems to work, but the filename isn't ok. 

`    filename = survey.description + '-' + survey.title +'.xlsx'

    writer = pd.ExcelWriter('./static/xlsfiles/'+filename, engine='xlsxwriter')

    all_votes = await statistics.get_all_values_for_download(survey_id)
    s = pd.DataFrame(all_votes)
    s.to_excel(writer, sheet_name='Sheet1')
    writer.save()

    return await send_from_directory('./static/xlsfiles/', file_name=filename)`

The result of this is almost ok, it gives a file named 'overview'. This is also the url of the page. 
It should be 'description-title.xlsx', that is also the value of filename when i print it....

Trying

` return await send_from_directory('./static/xlsfiles/', file_name=filename,  attachment_filename=filename)`
Gives an error that  **attachment_filename** is unexpected....

**Long story short:**
- is there a way to get it working using the bytesIO solution?
- if not, how can i set the attachment name ?

Your help would be greatly appreciated again...

kind regards

**Comment by pgjones** [comment]
I've added part of this functionality in 1c0003197a20979e35f32bfee938032edb1e9d91, namely the attachment part. I'll need to think about the IO implementation. Does it work for your use case?

**Comment by kllr** [comment]
Hi, great to hear you're thinking about implementing it! Writing to disk first and deleting it afterwards is a bit of a hassle.

For now it would be ok if i can rename the file to 'something.xlsx'. Right now it is 'overview.'. 
I probably can explain a work around for the users but if i can rename it, it would be much better. 

Is this possible do you think? (without the need for a new release of quart)

**Comment by pgjones** [comment]
That may work, you could also pin to a git commit in the short term.

**Comment by kllr** [comment]
Thanks, i'll look into it.
